### ISSUE TITLE
Subset Operator Implementation Query

### ISSUE THREAD

**Issue by nihesh** [ermongroup/subsets#1]
Hey,

From my understanding of the paper, we are interested in the top k items in the random key vector that is generated by performing Logit(weights) + Gumbel(0,1). If this is correct, why is the softmax computed over the "scores" tensor instead of the "gumbel_keys" tensor in line 73 of "subsets/subsets/knn/sorting_operator.py"?

**Comment by sangmichaelxie** [comment]
good catch. the code was changed during clean-up for publishing - this is fixed now.
